### PR TITLE
Fix #534: OIDC: Implement PKCE extension

### LINKED ISSUE #534: OIDC: Implement PKCE extension
PKCE will be implemented as an optional layer of security on top of OIDC activation.

Additional authorization code request parameters added by mobile application once PKCE is implemented:
- `code_challenge=CODE_CHALLENGE`
- `code_challenge_method=S256`

The server will need to obtain the code verifier value, so that the token can be received with PKCE protection.

The activation request will be updated to contain the `codeVerifier`.
```json
  "identityAttributes": {
      "method": "oidc",
      "clientId": "1234567890abcdef",
      "code": "1234567890abcdef",
      "nonce": "qfT4K8p7R5G3XY7zcm3HAw==",
      "codeVerifier": "G3hsI1KZX1o~K0p-5lT3F7yZ4bC8dE2jX9aQ6nO2rP3uS7wT5mV8jW1oY6xB3sD09tR4vU3qM1nG7kL6hV5wY2pJ0aF3eK9dQ8xN4mS2zB7oU5tL1cJ3vX6yP8rE2wO9n"
  }
```

Token request parameter added by the server when requesting a token with PKCE protection:
- `code_verifier=CODE_VERIFIER`

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Review by romanstrobl — APPROVED** [review]
Great!

File: powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/oidc/OidcHandler.java [hunk 1]
```diff
@@ -80,6 +80,7 @@ public String retrieveUserId(final OidcActivationContext request) throws PowerAu
 
         final TokenRequest tokenRequest = TokenRequest.builder()
                 .code(request.getCode())
+                .codeVerifier(request.getCodeVerifier())
```
Comment (zcgandcomp) [inline powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/oidc/OidcHandler.java:83]: If PKCE is enabled, it should be forced. I.e. if the code verifier is empty but PKCE is enabled by the config we should fail.

**Review by zcgandcomp — CHANGES_REQUESTED** [review]
We should enforce the PKCE if enabled in the config.

File: powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/oidc/OidcHandler.java (hunk 1 above)
Comment (banterCZ) [reply powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/oidc/OidcHandler.java:85]: Ok, let's force it.

**Review by zcgandcomp — APPROVED** [review]
Looks OK.

**Pull request merged**